### PR TITLE
Customizable FrontierMon Sets

### PR DESCRIPTION
**Pull request by ghoulslash**
Similar to the trainer control feature, this adds new fields to the FrontierMon struct such as shininess, pokeball type, evs, ivs for individual control, etc. The itemTableId field is replaced with a raw itemId. Vanilla frontier mon sets are updated accordingly.

Will Not Add:
ability for frontier mons to gigantimax/dynamax. Needs to be handled in battle_controller_opponent.c via shouldDynamax

### CODE REVIEW

**Comment by mrgriffin** [comment]
Forgive my ignorance, but is there a reason why we can't simply use `TrainerMon` instead of extending `FacilityMon`? Having them be literally the same `struct` would help people who write tools to generate 'mons, e.g. #3545.

**Comment by ghoulslash** [comment]
> Forgive my ignorance, but is there a reason why we can't simply use `TrainerMon` instead of extending `FacilityMon`? Having them be literally the same `struct` would help people who write tools to generate 'mons, e.g. #3545.

Mainly to save space. Nickname a few others are extraneous here. But I can merge them if that's easier

File: src/battle_tower.c [hunk 1]
```diff
@@ -1627,11 +1560,82 @@ static void FillTentTrainerParty(u8 monsCount)
...
+        
+        SetMonMoveSlot(dst, move, j);
+        if (gMovesInfo[move].effect == EFFECT_FRUSTRATION)
+            friendship = 0;  // Frustration is more powerful the lower the pokemon's friendship is.
+    }
+
+    SetMonData(dst, MON_DATA_FRIENDSHIP, &friendship);
+    SetMonData(dst, MON_DATA_HELD_ITEM, &fmon->heldItem);
+    
+    // try to set ability. Otherwise, random of non-hidden as per vanilla
```
Comment (AlexOn1ine) [inline src/battle_tower.c:1597]: The comment is a bit misleading. As far as I know it defaults to ability 0 but also it is what the code is doing here. I would just remove it.

**Comment by mrgriffin** [comment]
I'm not sure how to get GitHub to automatically close other issues, but I think this will close #3484 when it's done?

I'm busy until Thursday, so would it be okay for you to review this @AlexOn1ine? :)

**Review by AlexOn1ine — APPROVED** [review]
@mrgriffin hope there are no other concerns from you 

/Edit the comment was made yesterday before we decided to wait :sweat_smile:

**Pull request merged**